---
title: "ProcessVisCluster"
author: "Katie Rhodes"
date: "3/2/2021"
output: html_document
---

```{r}
library(Seurat)
```

load in filtered, normalized basal bone marrow count matrix
```{r}
dat<- read.table("/home/klrhodes/TusiTest/GSM2388072_basal_bone_marrow.filtered_normalized_counts.csv.gz", header=T, row.names=1, sep = ",")
```

check data structure
```{r}
dim(dat)
```

```{r}
head(dat)
```

```{r}
unique(dat$seq_run_id)
```
Two sequencing runs. In the paper methods they mention some batch effects likely due to sequencing depth.
```{r}
unique(dat$library_id)
```

```{r}
unique(dat$pass_filter)
```
These are the already filtered set of cells, so they all pass the filter

I believe genes have already been filtered as well.  lets make it a Seurat object and look at some QC metrics:

```{r}
#transpose to rows are genes and cells are columns
dat<- t(dat)
meta<- dat[1:4,]
meta<- as.data.frame(t(meta))

obj<- CreateSeuratObject(dat[5:nrow(dat),], meta.data = meta)
```

```{r}
head(obj)
```

check some QC metrics
```{r}
obj[['percent.mt']]<- PercentageFeatureSet(obj, pattern = '^mt.')
VlnPlot(obj, features = c("nCount_RNA", "nFeature_RNA", "percent.mt"), pt.size = 0.2)
```

So, cells have clearly been filtered for mt percentage (less than 10%). I also don't see obvious signs of doublets in the numbers of counts or features, but I'll check again after clustering. 

I believe these counts are already normalized, so proceed to finding variable features
```{r}
#subset to just the first sequencing run to find variable features
sub<- subset(obj, subset = seq_run_id == "seq_run1")
sub<- FindVariableFeatures(sub, selection.method = "vst", nfeatures=2000)
```

```{r}
#scaling data
all.genes<- rownames(obj)
obj<- ScaleData(obj, features=all.genes)
```

```{r}
#using only variable features from the first sequencing run
obj<- RunPCA(obj, features=VariableFeatures(sub))
```

```{r}
DimPlot(obj, reduction="pca")
```

```{r}
ElbowPlot(obj)
```

I'm gonna move forward with 15 PCs here

```{r}
obj<- FindNeighbors(obj, dims = 1:15)
obj<- FindClusters(obj, resolution = 0.5)
obj<- RunUMAP(obj, dims=1:15)
```
```{r}
DimPlot(obj, reduction="umap")
```

```{r}
DimPlot(obj, reduction="umap", group.by = "seq_run_id")
```

batch effect looks better!

and check if there might be a doublet cluster
```{r}
VlnPlot(obj, features = "nCount_RNA", pt.size =0)
```
certain clusters definitely have higher counts. This could be a real effect of cell type-- I'm going to leave it for now. 

now, visualizing marker gene expression
```{r}
#HSC marker
FeaturePlot(obj, features = c("Procr"))
```


```{r}
#erythroid markers
FeaturePlot(obj, features = c("Gata1", "Klf1"))
```

```{r}
#erythroid markers
FeaturePlot(obj, features = c("Epor", "Gypa"), ncol=2)
```

```{r}
#megakaryocyte markers
FeaturePlot(obj, features = c("Itga2b", "Pbx1"))
```

```{r}
#megakaryocyte markers
FeaturePlot(obj, features = c("Sdpr", "Vwf"))
```

```{r}
#Neutrophil and B cell markers
FeaturePlot(obj, features = c("Elane", "Cd19"))
```

```{r}
#basophil
FeaturePlot(obj, features = c("Mcpt8", "Prss34"))
```

So, this obviously doesn't look exactly like the UMAP from the paper, and there might even be cell types missiing because I didn't include the other samples (stressed, fetal liver, flow sorted). I don't know enough about these hematopoietic cells types to know whether I expect basophils to be more abundant in a certain condition, for example. 

but, I can see the erythroid branch and the small megakaryocyte branch, so I'll proceed from here. 

I'm ~1.5 hrs in at this point and haven't done formal trajectory inference (The plan was to convert to loom, move over to a jupyter notebook and run PAGA. If you guys want to see a previous jupyter notebook where I ran PAGA in the EB data, I can send that along!)

For the sake of time, I'll use the expected trajectories, and guess which cluster of cells would have been the branch point from erythroid to megakaryocyte. 

at clustering res 0.5, the megakaryocytes, progenitors, and early erythroid are all 1 cluster, so I need to try a higher resolution:

```{r}
obj<- FindClusters(obj, resolution = 1.5)
DimPlot(obj, label=T)
```
again check which of these clusters at higher res probably correspond to a cell type

without the trajectories, its tough to say which of these are the real branch point cluster. For now, I'm going to assume that based on the marker gene expression, cluster 6 and res 1.5 is multipotent progenitors, and I am going to assume that cluster 8 is the branch point between erythroid and megakaryocyte. I think this is probably flawed (I think there are a few basophils in cluster 8, the space of the UMAP where I might expect the erythroid/megakaryocyte progenitors to be is a little sparse...) but it'll work for now. 

```{r}
saveRDS(obj, "/home/klrhodes/TusiTest/PostProcessObj.Rmd")
```


